### ISSUE TITLE
Fail to run project/hn-sinc-nsf-9/00_demo.sh on Google Colaboratory

### ISSUE THREAD

**Issue by taroushirani** [nii-yamagishilab/project-NN-Pytorch-scripts#5]
Hello. I tried to run  project/hn-sinc-nsf-9/00_demo.sh on Google Colaboratory but  I failed. The contents of  log_err are as follows;

```
/usr/local/lib/python3.6/dist-packages/torch/functional.py:516: UserWarning: stft will require the return_complex parameter be explicitly  specified in a future PyTorch release. Use return_complex=False  to preserve the current behavior or return_complex=True to return  a complex output. (Triggered internally at  /pytorch/aten/src/ATen/native/SpectralOps.cpp:653.)
  normalized, onesided, return_complex)
Traceback (most recent call last):
  File "main.py", line 170, in <module>
    main()
  File "main.py", line 114, in main
    trn_set, val_set, checkpoint)
  File "/content/project-NN-Pytorch-scripts/core_scripts/nn_manager/nn_manager.py", line 336, in f_train_wrapper
    epoch_idx, optimizer, normtarget_f)
  File "/content/project-NN-Pytorch-scripts/core_scripts/nn_manager/nn_manager.py", line 153, in f_run_one_epoch
    loss_computed = loss_wrapper.compute(data_gen, normed_target)
  File "/content/project-NN-Pytorch-scripts/project/hn-sinc-nsf-9/model.py", line 957, in compute
    pad_mode="constant")
  File "/usr/local/lib/python3.6/dist-packages/torch/functional.py", line 516, in stft
    normalized, onesided, return_complex)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```
It seems that torch.hann_window in class Loss returns the Tensor on cpu, which is incompatible with outputs (on CUDA) at line 955 in  project/hn-sinc-nsf-9/model.py.   I uploaded the jupyter notebook file which I used to gist.github.com[1] so please see it for detail.

 1. https://gist.github.com/taroushirani/a263d1c33a23191ad0cedacf76f36409

**Comment by TonyWangX** [comment]
@taroushirani Thanks for pointing it out.

My mistake -- torch.window() should be used with dtype and device so that the tensor of the window can be created on device.  Please check the changed files in the linked pull request, https://github.com/nii-yamagishilab/project-NN-Pytorch-scripts/pull/6/files

It did not cause problem when window length != FFT_points, which is the case for all the model.py with default configurations.  In this case,  "window will be padded on both sides to length n_fft before being applied." https://pytorch.org/docs/stable/generated/torch.stft.html.  There, I guess that the padding operation will automatically convert the tensor to GPU. 

Anyway, I should have set dtype and device in torch.window().
Please have a try!

**Comment by taroushirani** [comment]
Thank you for your rapid response.  I tried the master-debugging branch and confirmed the problem is fixed.

I may have misunderstood your reply, but I used the hn-sinc-nsf-9/model.py with the default configulations and I met this problem.  I also tried this model several months ago but I did not encounter any problems, so I suspect that the difference of versions of pytorch(1.6.x->1.7.1) causes this problem.

Anyway the problem is solved. Thank you very much.
